src/util/bot-runner.js

```diff
@@ -36,6 +36,11 @@ export default class BotRunner {
     }
   }
 
+  // Find the first bot for which "fn" returns true.
+  findBot(fn) {
+    return Object.keys(this.bots).map(k => this.bots[k]).find(fn);
```

Comment:
not sure how well this would scale to a large quantity of bots--seems like we should be making an index by the key we care about for lookups, or bypassing the instantiated bot entirely, somehow.